soc/riscv/espressif_esp32/esp32c3/power.c

```diff
@@ -17,11 +17,6 @@ void pm_state_set(enum pm_state state, uint8_t substate_id)
 	ARG_UNUSED(substate_id);
 
 	switch (state) {
-	case PM_STATE_SOFT_OFF:
```

Comment:
isnt this going to remove policy based power control and moves the control to the application entirely? Some applications might want to do that, but is this always the case?

Reply:
I haven't seen cases where applications want the policy manager to decide when to power off the system based on timing constraints, at least on platforms I've ported. There's actually a collection of platform-specific samples in `samples/boards` where it's clear that SOFT_OFF was always used outside policy-based rules. On all platforms I'm familiar with, this mode typically implies configuring a wake-up source, then power off, so policy manager would not make much sense.

Reply:
Btw, on intel_adsp, it looks like SOFT_OFF does not imply state loss, because there are post ops:

https://github.com/zephyrproject-rtos/zephyr/blob/d11c0a1664f6052e68c6eebd35aa5aba88e7c290/soc/xtensa/intel_adsp/cavs/power.c#L117-L129

to me it looks like a misuse of SOFT_OFF:

https://github.com/zephyrproject-rtos/zephyr/blob/d11c0a1664f6052e68c6eebd35aa5aba88e7c290/include/zephyr/pm/state.h#L96-L106

But still... it's again used outside of policy:
https://github.com/zephyrproject-rtos/zephyr/blob/d11c0a1664f6052e68c6eebd35aa5aba88e7c290/dts/xtensa/intel/intel_adsp_ace20_lnl.dtsi#L59-L67

Reply:
> On all platforms I'm familiar with, this mode typically implies configuring a wake-up source, then power off, so policy manager would not make much sense.

you can enable the wake-up source and let the policy manager do the work, why this does not make sense? The wakeup source needs to be setup somewhere, whether you manually call power-off or you let the system PM do that is not different AFAIK.

Reply:
> Intel ADSP is more complex, the SRAM is powered off and all context is lost. The CPU is also power gated and lost context. What happens is that the context is stored in a external memory that is used when restarting it ... This is part of bigger platform where many decisions are made by the host CPU.

So then why do you have code in pm_state_exit_post_ops? How is this code reached?

Reply:
> > On all platforms I'm familiar with, this mode typically implies configuring a wake-up source, then power off, so policy manager would not make much sense.
> 
> you can enable the wake-up source and let the policy manager do the work, why this does not make sense? The wakeup source needs to be setup somewhere, whether you manually call power-off or you let the system PM do that is not different AFAIK.

You theoretically can, but nobody seems to be doing this. Instead, they don't define the state in DT, or add _infinite_ min-residency-us so that can be pm_state_force is used safely. To me, this is just a clunky API.